### ISSUE TITLE
Back button lag

### ISSUE THREAD

**Issue by DerVerruckteFuchs** [dularion/streama#410]
The back button is sometimes quite laggy. The lag lasts long enough that it seems to be completely unresponsive for quite a few seconds. However, allowing it to take it's time seems to work is rather consistently. After the lag it's quite responsive when loading the same video again. I find that it's pretty easily reproducible in Chromium when uploading an episode of a TV show, then playing it either from the Manage Content view or the Dashboard.

**Comment by dularion** [comment]
Do you mean the back button inside the video playback?

**Comment by DerVerruckteFuchs** [comment]
I think so. It's the one that takes you back to the dashboard by default.

**Comment by dularion** [comment]
Hm I cant seem to reproduce this currently. Can you maybe give me access to a reproducible case that I can debug?

**Comment by DerVerruckteFuchs** [comment]
What I'm working with right now is slightly different to master. I originally noticed the problem in 1.1, though it still seems present in 1.2.1_beta.
The diff of my git stash:

```
diff --git a/build.gradle b/build.gradle
index a517bf3..ddba7fa 100644
--- a/build.gradle
+++ b/build.gradle
@@ -11,7 +11,7 @@ buildscript {
         classpath "com.bertramlabs.plugins:asset-pipeline-gradle:2.6.4"
 //        classpath "gradle.plugin.com.craigburke.gradle:karma-gradle:1.4.3"
 //        classpath "gradle.plugin.com.craigburke.gradle:bower-installer-gradle:2.5.1"
-        classpath "org.grails.plugins:hibernate4:5.0.5"
+        classpath "org.grails.plugins:hibernate4:6.0.3"
         classpath "com.bertramlabs.plugins:asset-pipeline-gradle:2.8.2"
         classpath "org.grails.plugins:views-gradle:1.0.12"
     }
@@ -93,6 +93,10 @@ dependencies {
     compile "org.grails.plugins:mail:2.0.0.RC4"
     compile "org.grails.plugins:grails-spring-websocket:2.3.0"
     runtime 'mysql:mysql-connector-java:5.1.36'
+    compile group: 'org.postgresql', name: 'postgresql', version: '42.1.4'
+    compile 'org.grails.plugins:postgresql-extensions:5.1.0'
+    //compile 'org.hibernate:hibernate-core'
+
 }
 
 task wrapper(type: Wrapper) {
@@ -139,4 +143,4 @@ assets {
 bootRun {
     // local.config.location is just a random name. You can use yours.
     jvmArgs = ['-Dlocal.config.location=app-config.yml']
-}
\ No newline at end of file
+}
diff --git a/grails-app/domain/streama/User.groovy b/grails-app/domain/streama/User.groovy
index edd3d5b..42ee66e 100644
--- a/grails-app/domain/streama/User.groovy
+++ b/grails-app/domain/streama/User.groovy
@@ -36,6 +36,8 @@ class User {
        }
 
        static mapping = {
+               //User is reserved keyword in some systems, like PostgreSQL
+               table 'users'
                password column: '`password`'
                cache true
        }
```

I'm not sure if these changes are introducing any issues. I'm looking over log files to see if there is any clues there. Looking over the diff, the hibernate plugin is version 4:6.0.3. I'm going to revert that back to 4:5.0.5 to see if that changes anything.

**Comment by DerVerruckteFuchs** [comment]
I think the lag is a bit better after the 1.2.1_beta upgrade and reverting the change I did with the hibernate version. I am using Chromium for most of the troubleshooting. Firefox tends to be worse with the lag for some reason. I'm running Streama on a Debian 9 server, and accessing it on my desktop.

The more I test it, the more the reproducibility becomes inconsistent.
Here are the steps I take in Chromium:

- from the Dashboard select and play a video
- hit the back (return to Dashboard) button immediately or wait a few seconds
Then lag happens when trying to get back to the Dashboard. Sometimes when repeating for the same video, it only happens the first time. These last few tests it was happening consistently, then the lag stopped completely.

**Comment by dularion** [comment]
Hm very strange. This sounds like an angular issue and maybe even memory issue. Can you check the following: While you test open the chrome task manager (probably chromium has it also?) under Tools > Task manager. And then once the problem occurs check CPU and memory for that tab. I think something makes the cpu usage skyrocket in that moment.

**Comment by DerVerruckteFuchs** [comment]
I'm getting 4.0 to maybe 15.0 for CPU at most while paused and clicking the back button and waiting on the lag. When playing a video it's around 30.0. Memory usage seems pretty normal as well. When monitoring my server's performance everything seems mostly normal. CPU and memory usage is low and well within limits, there are quite a few processes for Streama but that's probably fine. All I can think of is loading back up the dash is slow sometimes for some reason. I'm wondering if it's a database access lag more than anything since the one major thing that's different is that I'm using postgres instead of mysql.

**Comment by DerVerruckteFuchs** [comment]
Is there an example Apache conf file to access Streama from a URL? I'm beginning to think my current Apache conf for my Streama URL, or some other Apache config somewhere, is responsible for the lag/slow loading of the Dashboard.

**Comment by Jeronimo95** [comment]
We only have details for Nginx: https://github.com/dularion/streama/wiki/Nginx-as-a-reverse-proxy

**Comment by DerVerruckteFuchs** [comment]
I think I've narrowed it down to http 2. Commenting out the `Protocols h2 http/1.1` line in my Apache config for Streama and restarting Apache eliminates any lag after hitting the back button. Getting back to the Dash is immediate. At first I thought it was DNS since testing Streama via the server IP instead of the URL eliminated the lag for me, but nothing I did with DNS seemed to do anything. There was another service on my server that had lots of trouble with http 2, so I tried that to see if that was also the case for Streama. Disabling http 2 for Streama revealed that was the problem.

I suppose better http 2 support may be a good idea in future. Perhaps this is something good to add to the wiki as an advisement for the time being.
